uvtools/plot.py

```diff
@@ -466,3 +469,265 @@ def omni_view_gif(filenames, name='omni_movie.gif'):
+              "fr" : "Fringe Rate [mHz]",
+              "dly" : "Delay [ns]",
+              }
+
+    # map plot types to transforms needed
+    plot_types = {"time_vs_freq" : lambda data : data, # do nothing
+                  "time_vs_dly" : lambda data : utils.FFT(data, 1), # FFT in freq
+                  "fr_vs_freq" : lambda data : utils.FFT(data, 0), # FFT in time
+                  "fr_vs_dly" : lambda data : utils.FFT(utils.FFT(data, 0), 1), # both
+                  }
+
+    # convert plot type to tuple
+    if isinstance(plot_type, str):
+        plot_type = tuple(plot_types.keys()) if plot_type == "all" else (plot_type,)
+
+    # check that chosen plot type(s) OK
+    assert all([plot in plot_types.keys() for plot in plot_type]), \
+            "Please ensure the plot type chosen is supported. The supported " \
+            "types are : {types}".format(types=list(plot_types.keys()))
+
+    # now make a dictionary of the transformed visibilities
+    visibilities = {plot : {label : xform(vis)
+                            for label, vis in zip(("vis1", "vis2"), (vis1, vis2))}
+                            for plot, xform in plot_types.items()
+                            if plot in plot_type} # but only use desired transforms
+
+
+    # import matplotlib, setup the figure
+    import matplotlib.pyplot as plt
+    figsize = (4 * 3, 3 * len(plot_type)) # (4,3) figsize for each plot
+    fig = plt.figure(figsize=figsize)
+    axes = fig.subplots(len(plot_type), 3)
+    axes = [axes,] if len(axes.shape) == 1 else axes # avoid bug for single row
+    axes[0][0].set_title("Amplitude Difference", fontsize=12)
+    axes[0][1].set_title("Phase Difference", fontsize=12)
+    axes[0][2].set_title("Amplitude of Complex Difference", fontsize=12)
+
+    # helper function for getting the extent of axes
+    extent = lambda xvals, yvals : (xvals[0], xvals[-1], yvals[-1], yvals[0])
+
+    # loop over items in visibilities and plot them
+    for i, item in enumerate(visibilities.items()):
+        # extract visibilities, get diffs
+        visA, visB = item[1].values()
+        diffs = (utils.diff(visA, visB, 'abs'), 
+                 utils.diff(visA, visB, 'phs'),
+                 utils.diff(visA, visB, 'complex'))
+
+        # extract parameters
+        ykey, xkey = item[0].split("_vs_") # keys for choosing parameters
+        xvals, yvals = plot_params[xkey], plot_params[ykey]
+
+        # get labels
+        xlabel, ylabel = labels[xkey], labels[ykey]
+
+        # plot stuff
+        for ax, diff in zip(axes[i], diffs):
+            # set labels
+            ax.set_xlabel(xlabel, fontsize=12)
+            ax.set_ylabel(ylabel, fontsize=12)
+
+            # plot waterfall and add a colorbar
+            fig.sca(ax)
+            cax = waterfall(diff, mode="real", cmap='viridis', 
+                            extent=extent(xvals, yvals))
+            fig.colorbar(cax)
+
+    # display the figure
+    plt.tight_layout()
+    plt.show()
```

Comment:
Best to let the user do this themselves, as otherwise it will make it harder to use this function in an automated way. In a Jupyter notebook, returning a `Figure` should result in the figure being plotted automatically anyway, so probably best to do that.